platform/default/mbgl/map/map_snapshotter.cpp

```diff
@@ -0,0 +1,76 @@
+#include <mbgl/map/map_snapshotter.hpp>
+
+#include <mbgl/actor/actor_ref.hpp>
+#include <mbgl/gl/headless_frontend.hpp>
+#include <mbgl/map/map.hpp>
+#include <mbgl/storage/file_source.hpp>
+#include <mbgl/style/style.hpp>
+#include <mbgl/util/event.hpp>
+
+namespace mbgl {
+
+class MapSnapshotter::Impl {
+public:
+    Impl(FileSource&,
+         Scheduler&,
+         const std::string& styleURL,
+         const Size&,
+         const float pixelRatio,
+         const CameraOptions&,
+         const optional<LatLngBounds> region,
+         const optional<std::string> programCacheDir);
+
+    void snapshot(ActorRef<MapSnapshotter::Callback>);
+
+private:
+    HeadlessFrontend frontend;
+    Map map;
+};
+
+MapSnapshotter::Impl::Impl(FileSource& fileSource,
+           Scheduler& scheduler,
+           const std::string& styleURL,
+           const Size& size,
+           const float pixelRatio,
+           const CameraOptions& cameraOptions,
+           const optional<LatLngBounds> region,
+           const optional<std::string> programCacheDir)
+    : frontend(size, pixelRatio, fileSource, scheduler, programCacheDir)
+    , map(frontend, MapObserver::nullObserver(), size, pixelRatio, fileSource, scheduler, MapMode::Still) {
+
+    map.getStyle().loadURL(styleURL);
+
+    map.jumpTo(cameraOptions);
+
+    // Set region, if specified
+    if (region) {
+        mbgl::EdgeInsets insets = { 0, 0, 0, 0 };
+        std::vector<LatLng> latLngs = { region->southwest(), region->northeast() };
+        map.jumpTo(map.cameraForLatLngs(latLngs, insets));
```

Comment:
Besides the coordinate, this also sets the pitch, etc. Since the map doesn't do anything significant before the render is started in `Still` mode I didn't see it as significant. (the onUpdate does not trigger an actual update yet).